### ISSUE TITLE
Can NCCL prevent cross NUMA communication, and use NIC to communicate instead?

### ISSUE THREAD

**Issue by yuxuanliuuu** [NVIDIA/nccl#930]
I have 2 groups of GPUs on both sides of NUMA.
However, due to the workload feature, it is desired to prevent cross-NUMA communication.  
Therefore, we want to connect the 2 groups of GPUs using NICs.

Is there a way to prevent cross NUMA communication, and use NIC instead in NCCL?
I really appreciate it if you can answer my question.

**Comment by KaimingOuyang** [comment]
Yes you can by setting `NCCL_SHM_DISABLE=1` and `NCCL_P2P_DISABLE=1`, however, this would force all communication going through NICs.

**Comment by sjeaugey** [comment]
By default, NCCL should already use the NIC to communicate across sockets, if NICs and GPUs are on a PCI switch. So you may not need to do anything.

If you want to force it, `NCCL_SHM_DISABLE=1` is the way to go on most platforms. `NCCL_P2P_DISABLE=1` should not be set; worst case you'd need to set `NCCL_P2P_LEVEL=PHB` so that we don't use P2P between sockets and revert to the NIC since SHM is disabled.

**Comment by yuxuanliuuu** [comment]
Thanks a lot！！！

**Comment by stephenmsachs** [comment]
### Issue Cleanup: Helping Us Focus on Current Challenges

  We're [reviewing](https://github.com/NVIDIA/nccl/discussions/1761) older issues to ensure we prioritize the most relevant and active ones. Since this issue hasn't seen updates in over 6 months, we'll be closing it for now.

  ,*This change helps us focus our efforts on addressing any current issues our users are facing.* If this issue still affects you, please don't hesitate to reopen it with a quick update (e.g., "Still relevant on [version=X]").
  Thanks for your understanding and for contributing to NCCL.
